rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf

```diff
@@ -583,10 +592,53 @@ SecRule ARGS_NAMES|ARGS|XML:/* "@rx (?i)[\s'\"`()]*?\b([\d\w]+)\b[\s'\"`()]*?(?:
     tag:'paranoia-level/2',\
     ver:'OWASP_CRS/3.4.0-dev',\
     severity:'CRITICAL',\
-    multiMatch,\
-    setvar:'tx.sql_injection_score=+%{tx.critical_anomaly_score}',\
-    setvar:'tx.anomaly_score_pl2=+%{tx.critical_anomaly_score}'"
+    setvar:'tx.942130_lhs=/%{TX.1}/',\
+    setvar:'tx.942130_matched_var_name=%{matched_var_name}',\
+    chain"
+    SecRule TX:942130_lhs "@within /%{TX.2}/" \
+        "t:none,\
+        setvar:'tx.sql_injection_score=+%{tx.critical_anomaly_score}',\
+        setvar:'tx.anomaly_score_pl2=+%{tx.critical_anomaly_score}'"
 
+# Rule Targeting logical inequalities that return TRUE (e.g. 1 != 2)
+#
+# The regexp written at the initial rule on the chain is the original prefix from the 942130.data.
+# When the prefix matches, we store the match to a new variable for comparing in the chained rule.
+#
+# We use captures to check for (in)equality in the regexp. So TX.1 will capture the left and side of the inequality,
+# and TX.2, TX.3 and TX.4 will capture the right hand side of the logical query.
```

Comment:
```suggestion
# and TX.2, TX.3 and TX.4 will capture the right hand side (RHS) of the logical query.
```

Reply:
Actually, the comment is no longer correct for the rule.

Reply:
Updating it.